doc/api/url.md

```diff
@@ -313,10 +313,21 @@ myURL.port = 1234.5678;
 console.log(myURL.port);
 // Prints 1234
 
-// Out-of-range numbers are ignored
+// Numbers which are represented in scientific notation in a String,
+// or as very large / small numbers in a Number,
+// will be assigned the first digit of the coefficient,
+// assuming the number is normalized (for example, 0.9e30 => 9e29).
+
+myURL.port = 4.567e21;
+console.log(myURL.port);
+// Prints 4 (because the coefficient is 4.567)
+
+// Out-of-range numbers, which are not represented in scientific noation,
+// will be ignored.
 myURL.port = 1e10;
```

Comment:
I guess it isn't super clear that `(1e10).toString()` is `'10000000000'` -- might be good to clarify that.

It got me thinking that it might be good to just describe the actual algorithm for assigning a number to `port` in the prose (ll. 322-329): **the number is first stringified using `String(num)`**, and then the leading characters of the resulting string that are digits are then converted back to a number. (Note: in cases where the default representation of the number is an exponential, the parts after the decimal point or the `e` character are discarded.) Only then is the range checked.

Reply:
I've rephrased the whole thing, adopting most of your advice, and it is much clearer now in my opinion.